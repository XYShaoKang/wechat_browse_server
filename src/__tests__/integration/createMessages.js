import { Prisma, graphqlTestCall } from '../__utils'
import { CreateMessages } from '../__types'
import { messageData } from '../__variables'

describe('Mutation', () => {
  /** @type {Prisma} */
  let prisma
  beforeEach(() => {
    prisma = new Prisma()
  })
  it('create Messages', async () => {
    await graphqlTestCall({
      query: CreateMessages,
      variables: messageData,
      context: {
        prisma,
        currentUser: { id: '0' },
        currentWeChat: { id: '0' },
      },
    })

    expect(prisma.updateWeChat).toHaveBeenCalled()
    expect(prisma.createMessage).toHaveBeenCalled()
    expect(prisma.connectWeChatUser).toHaveBeenCalled()
    expect(prisma.connectChatRoom).toHaveBeenCalled()
    expect(prisma.db.messages).toMatchSnapshot()
  })
  it('create Messages Fail No WeChat', async () => {
    const result = await graphqlTestCall({
      query: CreateMessages,
      variables: messageData,
      context: {
        prisma,
        currentUser: { id: '0' },
      },
    })

    const message = result.errors && result.errors[0].message
    expect(message).toBe('No WeChat')
    expect(result).toMatchSnapshot()
  })
})
