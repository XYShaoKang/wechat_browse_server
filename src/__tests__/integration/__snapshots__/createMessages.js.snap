// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation create Messages 1`] = `
Array [
  Object {
    "chatRoom": null,
    "createTime": 2016-05-19T00:00:22.935Z,
    "id": "0",
    "isSend": 0,
    "msgSvrId": "1000000000000000001",
    "talker": Object {
      "id": "0",
    },
    "type": 1,
  },
  Object {
    "chatRoom": Object {
      "id": "0",
    },
    "createTime": 2016-05-19T00:00:22.935Z,
    "id": "1",
    "isSend": 1,
    "msgSvrId": "1000000000000000002",
    "talker": Object {
      "id": "1",
    },
    "type": 1,
  },
  Object {
    "chatRoom": Object {
      "id": "0",
    },
    "createTime": 2016-07-10T11:40:28.809Z,
    "id": "2",
    "isSend": 0,
    "msgSvrId": "1000000000000000003",
    "talker": null,
    "type": 1,
  },
]
`;

exports[`Mutation create Messages Fail No WeChat 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: No WeChat],
  ],
}
`;
